ssc_interface_wrapper_ros2/src/converter_worker.cpp

```diff
@@ -188,88 +205,104 @@ namespace ssc_interface_wrapper{
+            // steering angle [rad]
+            vehicle_status.angle = std::atan(curvature * config_.wheel_base_);
 
-        // gearshift
-        if (msg_gear.current_gear.gear == automotive_platform_msgs::msg::Gear::NONE)
-        {
-            vehicle_status.current_gear.gear = automotive_platform_msgs::msg::Gear::NONE;
-        }
-        else if (msg_gear.current_gear.gear == automotive_platform_msgs::msg::Gear::PARK)
-        {
-            vehicle_status.current_gear.gear = automotive_platform_msgs::msg::Gear::PARK;
-        }
-        else if (msg_gear.current_gear.gear == automotive_platform_msgs::msg::Gear::REVERSE)
-        {
-            vehicle_status.current_gear.gear= automotive_platform_msgs::msg::Gear::REVERSE;
-        }
-        else if (msg_gear.current_gear.gear == automotive_platform_msgs::msg::Gear::NEUTRAL)
-        {
-            vehicle_status.current_gear.gear = automotive_platform_msgs::msg::Gear::NEUTRAL;
-        }
-        else if (msg_gear.current_gear.gear == automotive_platform_msgs::msg::Gear::DRIVE)
-        {
-            vehicle_status.current_gear.gear = automotive_platform_msgs::msg::Gear::DRIVE;
-        }
+            // gearshift
+            if (msg_gear.current_gear.gear == automotive_platform_msgs::msg::Gear::NONE)
+            {
+                vehicle_status.current_gear.gear = automotive_platform_msgs::msg::Gear::NONE;
+            }
+            else if (msg_gear.current_gear.gear == automotive_platform_msgs::msg::Gear::PARK)
+            {
+                vehicle_status.current_gear.gear = automotive_platform_msgs::msg::Gear::PARK;
+            }
+            else if (msg_gear.current_gear.gear == automotive_platform_msgs::msg::Gear::REVERSE)
+            {
+                vehicle_status.current_gear.gear= automotive_platform_msgs::msg::Gear::REVERSE;
+            }
+            else if (msg_gear.current_gear.gear == automotive_platform_msgs::msg::Gear::NEUTRAL)
+            {
+                vehicle_status.current_gear.gear = automotive_platform_msgs::msg::Gear::NEUTRAL;
+            }
+            else if (msg_gear.current_gear.gear == automotive_platform_msgs::msg::Gear::DRIVE)
+            {
+                vehicle_status.current_gear.gear = automotive_platform_msgs::msg::Gear::DRIVE;
+            }
+
+            // lamp/light cannot be obtain from SSC
+            // vehicle_status.lamp
+            // vehicle_status.light
 
-        // lamp/light cannot be obtain from SSC
-        // vehicle_status.lamp
-        // vehicle_status.light
+            current_status_msg_ = vehicle_status;
+            have_vehicle_status_ = true; // Set vehicle status message flag to true
 
-        current_status_msg_ = vehicle_status;
-        have_vehicle_status_ = true; // Set vehicle status message flag to true
+            // Reset flags
+            received_new_msg_ = false;
+            single_update_processed_ = !single_update_processed_;
```

Comment:
I'm not really clear on what these two flags do, 
Seems like you would want to set 
```
velocity_msg_exists_ , curvature_msg_exists_ , throttle_msg_exists_ , brake_msg_exists_ , gear_msg_exists_ , steering_msg_exists_
```
To false here so that you can receive a new set of updates and then publish that